### PR TITLE
AEPTarget API reference and migration docs

### PR DESCRIPTION
**Pull request by rymorale**

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -70,10 +70,13 @@
   * [Migration to AEPPlaces v2](foundation-extensions/places/migration-v2.md)
   * [Places API Reference](foundation-extensions/places/places-usage-reference.md)
   * [ACPPlaces \(iOS\) Reference](foundation-extensions/places/adobe-places.md)
-* [Profile](foundation-extensions/profile/README.md)
-  * [Migrating to AEPUserProfile](foundation-extensions/profile/acpuserprofile-aepuserprofile.md)
+* [Adobe Experience Platform Profile](foundation-extensions/profile/README.md)
+  * [Migration to AEPUserProfile](foundation-extensions/profile/acpuserprofile-aepuserprofile.md)
   * [Profile API reference](foundation-extensions/profile/profile-api-references.md)
   * [Profiles event reference](foundation-extensions/profile/profiles-event-reference.md)
+* [Adobe Experience Platform Target](foundation-extensions/target/README.md)
```
Comment (swarna04) [inline SUMMARY.md:77]: Do all Swift extensions go under `Mobile Foundations`? AJO is under `Experience Cloud`. How are we deciding the sections?

File: foundation-extensions/target/migration.md [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+{% tab title="AEPTarget (Objective-C)" %}
+
+```objc
++ (void) clearPrefetchCache;
+```
+
+{% endtab %}
+
+{% tab title="AEPTarget (Swift)" %}
```
Comment (swarna04) [inline foundation-extensions/target/migration.md:89]: I suppose we are following the order 
`AEP 3.x (Swift) | AEP 3.x (Objective-C) | ACP 2.x (Objective-C)`

File: foundation-extensions/target/migration.md [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+- [displayedLocations](#displayedLocations)
+- [extensionVersion](#extensionVersion)
+- [getThirdPartyId](#getThirdPartyId)
+- [getTntId](#getTntId)
+- [prefetchContent](#prefetchContent)
+- [registerExtension](#registerExtension)
+- [resetExperience](#resetExperience)
+- [retrieveLocationContent](#retrieveLocationContent)
+- [setPreviewRestartDeepLink](#setPreviewRestartDeepLink)
+- [setThirdPartyId](#setThirdPartyId)
```
Comment (swarna04) [inline foundation-extensions/target/migration.md:66]: We decided to remove this TOC. One can use the right nav bar for navigating APIs.

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: SUMMARY.md (hunk 1 above)
Comment (swarna04) [reply SUMMARY.md:77]: Also, now Target API reference is in two sections. What's our plan here?

File: foundation-extensions/target/migration.md [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+{% endtab %}
+{% endtabs %}
+
+## Primary class
+
+The class name containing public APIs is different depending on which SDK and language combination being used.
+
+| SDK Version | Language    | Class Name        | Example                                 |
+| ----------- | ----------- | ----------------- | --------------------------------------- |
+| ACPTarget   | Objective-C | `ACPCTarget`      | `[ACPTarget clearPrefetchCache];`       |
```
Comment (swarna04) [inline foundation-extensions/target/migration.md:48]: Typo `ACPCTarget`

File: foundation-extensions/target/migration.md [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+
+| AEP (3.x)       | ACP (2.x) |
+| :---------------- | :-------------------------------------- |
+| AEPMobileTarget | ACPTarget                        |
+| AEPMobileCore     | ACPCore                                 |
+
+{% endtab %}
+{% endtabs %}
+
+## Primary class
```
Comment (swarna04) [inline foundation-extensions/target/migration.md:42]: Why do you have two separate sections, refer to L20 and L42. Shouldn't we merge these? Also I do not see a reference to other public classes e.g. TargetRequest, TargetPrefetch, TargetParameters etc

File: foundation-extensions/target/target-api-reference.md [hunk 6]
```diff
@@ -0,0 +1,689 @@
...
+- [retrieveLocationContent](#retrieveLocationContent)
+- [setThirdPartyId](#setThirdPartyId)
+- [getThirdPartyId](#getThirdPartyId)
+- [getTntId](#getTntId)
+- [resetExperience](#resetExperience)
+- [clearPrefetchCache](#clearPrefetchCache)
+- [setPreviewRestartDeepLink](#setPreviewRestartDeepLink)
+- [displayedLocations](#displayedLocations)
+- [clickedLocation](#clickedLocation)
+- [collectLaunchInfo (For enabling visual preview mode)](#Visual preview)
```
Comment (swarna04) [inline foundation-extensions/target/target-api-reference.md:19]: We can remove this TOC in favor of right navbar for APIs

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: foundation-extensions/target/migration.md (hunk 2 above)
Comment (rymorale) [reply foundation-extensions/target/migration.md:89]: I am redoing the migration docs to follow the format that AEPPlaces is using: https://aep-sdks.gitbook.io/docs/foundation-extensions/places/migration

File: foundation-extensions/target/migration.md (hunk 5 above)
Comment (rymorale) [reply foundation-extensions/target/migration.md:42]: This will be fixed in the migration format change.

File: SUMMARY.md (hunk 1 above)
Comment (rymorale) [reply SUMMARY.md:77]: What do you think of adding a link to the ACPTarget docs? Or is this too confusing?

**Review by calebk1 — COMMENTED** [review]
Just a few comments!

File: foundation-extensions/target/migration.md [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+
+For example:
+
+{% hint style="info" %}
+This is information that is important to help clarify the API.
+{% endhint %}
+
+## Pod installation
+
+| AEP (3.x)                    | ACP (2.x) |
```
[the remark this answers is not in the document]
Comment (rymorale) [reply foundation-extensions/target/migration.md]: I have just pushed a commit (probably the same time you were reviewing this) that updates the migration doc to use tabs.

File: foundation-extensions/target/migration.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rymorale) [reply foundation-extensions/target/migration.md]: This is covered in my latest commit as well.

File: foundation-extensions/target/target-api-reference.md [hunk 8]
```diff
@@ -0,0 +1,689 @@
...
+
+```objc
+[AEPMobileCore registerExtensions:@[AEPMobileTarget.class, ...] completion:^{
+  // processing after registration
+}];
+```
+
+{% endtab %}
+
+{% endtabs %}
```
[the remark this answers is not in the document]
Comment (rymorale) [reply foundation-extensions/target/target-api-reference.md:76]: I think that sounds good. I'll remove it and just point to the MobileCore registerExtensions api docs.

File: foundation-extensions/target/migration.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (calebk1) [reply foundation-extensions/target/migration.md]: Yeah, I got an outdated commit and then I couldn't find my comment here, so I couldn't delete this comment 😅

[the remark this answers is not in the document]
Comment (rymorale) [reply foundation-extensions/target/migration.md]: just realized I updated the migration docs following the format which was *NOT* selected :(. i'll have to do an update for the target and campaign migration docs.

**Comment by rymorale** [comment]
Hi @calebk1, these are ready for a quick re-review as well. Thanks!

**Review by calebk1 — COMMENTED** [review]
Just a few more comments.

File: foundation-extensions/target/migration.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+| Type                   | AEP (3.x)        | ACP (2.x)               |
+| ---------------------- | :--------------- | :---------------------- |
+| Primary Class (Module) | Target           | ACPTarget               |
+| Class                  | TargetRequest    | ACPTargetRequestObject  |
+| Class                  | TargetPrefetch   | ACPTargetPrefetchObject |
+| Class                  | TargetOrder      | ACPTargetOrder          |
+| Class                  | TargetParameters | ACPTargetParameters     |
+| Class                  | TargetProduct    | ACPTargetProduct        |
+
+## Public APIs \(alphabetical\)
```
[the remark this answers is not in the document]
Comment (rymorale) [reply foundation-extensions/target/migration.md:18]: good point, updating

**Review by calebk1 — APPROVED** [review]
LGTM! Just one comment, and we'd be good to merge.

File: SUMMARY.md [hunk 10]
```diff
@@ -69,10 +69,13 @@
   * [Migration to AEPPlaces](foundation-extensions/places/migration.md)
   * [Places API Reference](foundation-extensions/places/places-usage-reference.md)
   * [ACPPlaces \(iOS\) Reference](foundation-extensions/places/adobe-places.md)
-* [Profile](foundation-extensions/profile/README.md)
-  * [Migrating to AEPUserProfile](foundation-extensions/profile/acpuserprofile-aepuserprofile.md)
+* [Adobe Experience Platform Profile](foundation-extensions/profile/README.md)
```
Comment (emdobrin) [inline SUMMARY.md:72]: I am afraid this will be confused with AEP User Profile, can we keep this Profile?

File: SUMMARY.md [hunk 11]
```diff
@@ -69,10 +69,13 @@
...
+  * [Migration to AEPUserProfile](foundation-extensions/profile/acpuserprofile-aepuserprofile.md)
```
Comment (emdobrin) [inline SUMMARY.md:73]: We should use Migrating to <extension name> reference for all extensions

File: SUMMARY.md (hunk 1 above)
Comment (emdobrin) [reply SUMMARY.md:76]: I would say this should go under Experience Cloud which captures the solution extensions.

**Comment by rymorale** [comment]
Hi @ahmyr and @praschetan, is this pr good to merge?

File: using-mobile-extensions/adobe-target/migration.md [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+```swift
+static func getThirdPartyId(_ completion: @escaping (String?, Error?) -> Void)
+```
+
+{% endtab %}
+
+{% tab title="AEP 3.x (Objective-C)" %}
+
+```objc
++ (void) getThirdPartyId:^(NSString* _Nullable, NSError* _Nullable) completion;
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/migration.md:120]:
```
+ (void) getThirdPartyId: (nonnull void (^) (NSString* _Nullable thirdPartyId,  NSError* _Nullable error)) completion;
```

File: using-mobile-extensions/adobe-target/migration.md [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+```swift
+static func getTntId(_ completion: @escaping (String?, Error?) -> Void)
+```
+
+{% endtab %}
+
+{% tab title="AEP 3.x (Objective-C)" %}
+
+```objc
++ (void) getTntId:^(NSString* _Nullable, NSError* _Nullable) completion;
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/migration.md:148]:
```
+ (void) getTntId: (nonnull void (^) (NSString* _Nullable tntId,  NSError* _Nullable error)) completion;
```

File: using-mobile-extensions/adobe-target/migration.md [hunk 14]
```diff
@@ -0,0 +1,312 @@
...
+```
+
+{% endtab %}
+
+{% tab title="AEP 3.x (Objective-C)" %}
+
+```objc
++ (void) prefetchContent: (NSArray<AEPTargetPrefetchObject*>* _NonNull)
+          withParameters: (AEPTargetParameters* _Nullable)
+                callback: ^(NSError* _Nullable) completion;
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/migration.md:178]:
```
+ (void) prefetchContent: (NSArray<AEPTargetPrefetchObject*>* _NonNull) prefetchArray
               withParameters: (AEPTargetParameters* _Nullable) targetParameters
                            callback: (nullable void (^) (NSError* _Nullable error)) completion;
```

File: using-mobile-extensions/adobe-target/migration.md [hunk 15]
```diff
@@ -0,0 +1,312 @@
...
+[AEPMobileCore registerExtensions:@[AEPMobileTarget.class] completion:nil];
+```
+
+{% endtab %}
+
+{% tab title="ACP 2.x (Objective-C)" %}
+
+```objc
++ (void) registerExtension;
+```
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/migration.md:223]: Here in first two tabs for 3.x you are providing the example usage and in the third tab for 2.x syntax, can we fix this please?

File: using-mobile-extensions/adobe-target/migration.md [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+static func retrieveLocationContent(_ requestArray: [TargetRequest], with targetParameters: TargetParameters? = nil)
+```
+
+{% endtab %}
+
+{% tab title="AEP 3.x (Objective-C)" %}
+
+```objc
++ (void) retrieveLocationContent: (NSArray<AEPTargetRequestObject*>* _NonNull)
+                  withParameters: (AEPTargetParameters* _Nullable);
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/migration.md:243]:
```
 (void) retrieveLocationContent: (NSArray<AEPTargetRequestObject*>* _NonNull) requestsArray
                          withParameters: (AEPTargetParameters* _Nullable) targetParameters;
```

File: using-mobile-extensions/adobe-target/migration.md [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+```swift
+static func setPreviewRestartDeepLink(_ deeplink: URL)
+```
+
+{% endtab %}
+
+{% tab title="AEP 3.x (Objective-C)" %}
+
+```objc
++ (void) setPreviewRestartDeeplink: (NSURL* _NonNull);
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/migration.md:272]:
```
+ (void) setPreviewRestartDeeplink: (NSURL* _NonNull) deeplink;
```

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: using-mobile-extensions/adobe-target/migration.md [hunk 18]
```diff
@@ -0,0 +1,312 @@
+# Migrating to AEPTarget reference
+
+This document is a reference comparison of ACPTarget (2.x) APIs against their equivalent APIs in AEPTarget (3.x) for an iOS mobile application implementation.
+
+The AEPTarget extension is implemented purely in Swift and is compatible with the AEPCore swift SDK. To ensure a smooth transition from the ACPTarget SDK, there are no major changes on the API names or definition. For more details, follow the migration guide below for your Swift or Objective-C mobile application.  If explanation beyond showing API differences is necessary, it will be captured as an info hint within that API's section.
+
+## AEPTarget classes
```
[the remark this answers is not in the document]
Comment (rymorale) [reply using-mobile-extensions/adobe-target/migration.md:7]: This naming format is used by the other migration docs as well. I think we should keep it as is to be uniform. what do you think?

[the remark this answers is not in the document]
Comment (rymorale) [reply using-mobile-extensions/adobe-target/migration.md:7]: this makes sense to change this heading to Public classes and the following to Public APIs. It looks like the other migration docs are using the AEPExtension classes and AEPExtension APIs heading. They should probably follow the format you suggested.

File: using-mobile-extensions/adobe-target/target-api-reference.md [hunk 19]
```diff
@@ -1,591 +1,755 @@
...
-
-{% hint style="warning" %}
-Do not use this API with the deprecated [loadRequests](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-target/target-api-reference-deprecated#load-target-requests) API. For a prefetched mbox, after calling the deprecated `loadRequests` API, the mbox display notification is sent internally to the Target server by the SDK.
-{% endhint %}
+---
+### getTntId
+This API gets the Target user ID (also known as the `tntId`) from the Target service. The `tntId` is returned in the network response after a successful call to `prefetchContent`, which is then persisted in the SDK. This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall or when the `resetExperience` API is used.
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/target-api-reference.md:629]: The tntId is returned in the response for both `prefetchContent` and `retrieveLocationContent`. Can we fix the description please?

**Comment by swarna04** [comment]
@rymorale The migration doc updates look good. I've provided some feedback on API reference doc which I had not reviewed before. Also, I noticed we removed all public classes in the API reference. I'm assuming we are following this across the board and the change is not specific to this extension.

**Review by swarna04 — CHANGES_REQUESTED** [review]

**Comment by rymorale** [comment]
> @rymorale The migration doc updates look good. I've provided some feedback on API reference doc which I had not reviewed before. Also, I noticed we removed all public classes in the API reference. I'm assuming we are following this across the board and the change is not specific to this extension.

I took a look at the Places docs (https://aep-sdks.gitbook.io/docs/foundation-extensions/places/places-usage-reference#additional-classes-and-enums) and Core (https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#public-classes) and I see a mix of public class info. I think I will restore in the Target API reference doc and update it to include AEPTarget classes.

File: using-mobile-extensions/adobe-target/target-api-reference.md [hunk 20]
```diff
@@ -1769,3 +1769,717 @@ ACPCore.collectLaunchInfo(["adb_deeplink" : "com.adobe.targetpreview://app.adobe
...
+     * @param targetParameters TargetParameters for this prefetch request
+     */
+     public TargetPrefetch(final String mboxName, final TargetParameters targetParameters)
+
+     /**
+     * Sets mbox parameters for the request.
+     *
+     * @param mboxParameters Map<String, String> mbox parameters
+     */
+     void setMboxParameters(final Map<String, String> mboxParameters);
```
Comment (swarna04) [inline using-mobile-extensions/adobe-target/target-api-reference.md]: In the Java classes, should we only specify the public methods?

**Review by swarna04 — CHANGES_REQUESTED** [review]
Provided some more feedback (minor), please feel free to address as needed. You can merge the PR once done.

File: using-mobile-extensions/adobe-target/target-api-reference.md (hunk 20 above)
Comment (rymorale) [reply using-mobile-extensions/adobe-target/target-api-reference.md]: i agree, i'll make this change

**Pull request merged**